MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -38,211 +40,146 @@ Initialise the Interface
-          this, SLOT(showContextMenu(const QPoint &)));
   connect(ui.tableSearchResults,
           SIGNAL(customContextMenuRequested(const QPoint &)), this,
           SLOT(showSearchContextMenu(const QPoint &)));
-  // Finally, create a presenter to do the thinking for us
+  // Synchronize the two instrument selection widgets
+  connect(ui.comboSearchInstrument, SIGNAL(currentIndexChanged(int)),
+          ui.qReflTableView,
+          SLOT(on_comboProcessInstrument_currentIndexChanged(int)));
+  connect(ui.qReflTableView,
+          SIGNAL(comboProcessInstrument_currentIndexChanged(int)),
+          ui.comboSearchInstrument, SLOT(setCurrentIndex(int)));
+
+  // Needed to Import/Export TBL, plot row and plot group
+  connect(ui.qReflTableView, SIGNAL(runAsPythonScript(const QString &, bool)),
+          this, SIGNAL(runAsPythonScript(const QString &, bool)));
+
   m_presenter = boost::make_shared<ReflMainViewPresenter>(
-      this /*main view*/,
+      this /*main view*/, ui.qReflTableView->getTablePresenter().get(),
       this /*currently this concrete view is also responsibile for prog reporting*/);
   m_algoRunner = boost::make_shared<MantidQt::API::AlgorithmRunner>(this);
 }
 
 /**
-This slot loads a table workspace model and changes to a LoadedMainView
-presenter
-@param name : the string name of the workspace to be grabbed
-*/
-void QtReflMainView::setModel(QString name) {
-  m_toOpen = name.toStdString();
-  m_presenter->notify(IReflPresenter::OpenTableFlag);
-}
-
-/**
-* Set all possible tranfer methods
-* @param methods : All possible transfer methods.
+* Add a command (action) to a menu
+* @param menu : [input] The menu where actions will be added
+* @param command : [input] The command (action) to add
 */
-void QtReflMainView::setTransferMethods(const std::set<std::string> &methods) {
-  for (auto method = methods.begin(); method != methods.end(); ++method) {
-    ui.comboTransferMethod->addItem((*method).c_str());
-  }
-}
+void QtReflMainView::addToMenu(QMenu *menu, ReflCommandBase_uptr command) {
 
-/**
-Set a new model in the tableview
-@param model : the model to be attached to the tableview
-*/
-void QtReflMainView::showTable(QReflTableModel_sptr model) {
-  m_model = model;
-  // So we can notify the presenter when the user updates the table
-  connect(m_model.get(),
-          SIGNAL(dataChanged(const QModelIndex &, const QModelIndex &)), this,
-          SLOT(tableUpdated(const QModelIndex &, const QModelIndex &)));
-  ui.viewTable->setModel(m_model.get());
-  ui.viewTable->resizeColumnsToContents();
-  std::string windowTitle = "ISIS Reflectometry (Polref) - " + m_toOpen;
-  auto mainWindowWidget = this->topLevelWidget();
-  mainWindowWidget->setWindowTitle(QString::fromStdString(windowTitle + "[*]"));
-  this->setWindowModified(false);
+  // cppcheck-suppress unreadVariable
+  ReflCommandAdapter *adapter =
+      new ReflCommandAdapter(menu, std::move(command));
```

Comment:
I think the only solution is to store these adapters as member functions somehow. Definitely as `unique_ptr`, I think in `vectors` of `unique_ptr` makes the most sense.